pkg/ruler/manager.go

```diff
@@ -309,9 +312,27 @@ func (r *DefaultMultiTenantManager) getOrCreateNotifier(userID string, userManag
 			defer sp.Finish()
 			ctx = ot.ContextWithSpan(ctx, sp)
 			_ = ot.GlobalTracer().Inject(sp.Context(), ot.HTTPHeaders, ot.HTTPHeadersCarrier(req.Header))
-			return ctxhttp.Do(ctx, client, req)
+			resp, err := ctxhttp.Do(ctx, client, req)
+			if err != nil {
+				level.Error(logger).Log("msg", "error occurred while sending alerts", "error", err)
+				return resp, err
+			}
+			defer resp.Body.Close()
+			if resp.StatusCode/100 != 2 {
+				bodyBytes, err := io.ReadAll(resp.Body)
+				if err != nil {
+					level.Error(logger).Log("msg", "error reading response body", "error", err, "response code", resp.StatusCode)
+					return resp, err
+				}
+				customErrorMessage := string(bodyBytes)
+				if len(customErrorMessage) >= 150 {
+					customErrorMessage = customErrorMessage[:150]
```

Comment:
so technically, you log the first 150 characters of unsuccessful calls to alertmanager?

Reply:
Correct. This is just to ensure if alert manager returns a very long message, we log the first 150 characters